engine/classes/Elgg/Application.php

```diff
@@ -67,145 +72,170 @@ class Application {
+			$config->$name = $value;
+			$lock($name);
+		}
+
+		// move sensitive credentials into isolated services
+		$this->services->dbConfig;
+
+		// move sensitive credentials into isolated services
+		$secret = SiteSecret::fromConfig($config);
+		if ($secret) {
+			$this->services->setValue('siteSecret', $secret);
+		}
+
+		$config->boot_complete = false;
 	}
 
 	/**
-	 * Load settings.php
+	 * Get the DB credentials.
 	 *
-	 * This is done automatically during the boot process or before requesting a database object
+	 * We no longer leave DB credentials in the config in case it gets accidentally dumped.
 	 *
-	 * @see Config::loadSettingsFile
-	 * @return void
+	 * @return \Elgg\Database\DbConfig
 	 */
-	public function loadSettings() {
-		$this->services->config->loadSettingsFile();
+	public function getDbConfig() {
+		return $this->services->dbConfig;
 	}
 
 	/**
-	 * Load all Elgg procedural code and wire up boot events, but don't boot
+	 * Define all Elgg global functions and constants, wire up boot events, but don't boot
 	 *
-	 * This is used for internal testing purposes
+	 * This includes all the .php files in engine/lib (not upgrades). If a script returns a function,
+	 * it is queued and executed at the end.
 	 *
 	 * @return void
 	 * @access private
 	 * @internal
+	 * @throws \InstallationException
 	 */
 	public function loadCore() {
-		if (function_exists('elgg')) {
+		if (self::$core_loaded) {
 			return;
 		}
 
-		$lib_dir = self::elggDir()->chroot("engine/lib");
-
-		// load the rest of the library files from engine/lib/
-		// All on separate lines to make diffs easy to read + make it apparent how much
-		// we're actually loading on every page (Hint: it's too much).
-		$lib_files = [
-			// Needs to be loaded first to correctly bootstrap
-			'autoloader.php',
-			'elgglib.php',
-
-			// The order of these doesn't matter, so keep them alphabetical
-			'access.php',
-			'actions.php',
-			'admin.php',
-			'annotations.php',
-			'cache.php',
-			'comments.php',
-			'configuration.php',
-			'cron.php',
-			'database.php',
-			'entities.php',
-			'filestore.php',
-			'group.php',
-			'input.php',
-			'languages.php',
-			'mb_wrapper.php',
-			'memcache.php',
-			'metadata.php',
-			'metastrings.php',
-			'navigation.php',
-			'notification.php',
-			'objects.php',
-			'output.php',
-			'pagehandler.php',
-			'pageowner.php',
-			'pam.php',
-			'plugins.php',
-			'private_settings.php',
-			'relationships.php',
-			'river.php',
-			'sessions.php',
-			'sites.php',
-			'statistics.php',
-			'system_log.php',
-			'tags.php',
-			'user_settings.php',
-			'users.php',
-			'upgrade.php',
-			'views.php',
-			'widgets.php',
-
-			// backward compatibility
-			'deprecated-2.1.php',
-			'deprecated-3.0.php',
-		];
+		$setups = [];
+		$path = Paths::elgg() . 'engine/lib';
 
-		// isolate global scope
-		call_user_func(function () use ($lib_dir, $lib_files) {
+		$files = scandir($path, SCANDIR_SORT_ASCENDING);
```

Comment:
I'll change it but I think it's a bad idea to support people leaving old code files lying around. This can easily break things like views and leave unsupported actions.